### PR TITLE
Update hoist tests to use in-repo run.sh

### PR DESCRIPTION
**Pull request by gandelman-a**
Lets hold off on merging this till all of our zuul changes have been picked up by the production server and I've confirmed a similar migration on BonnyCI/sandbox

### CODE REVIEW

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.com/check_github/BonnyCI/hoist/103/1486146609.58/ubuntu-hoist : SUCCESS in 4m 19s
- http://logs.bonnyci.com/check_github/BonnyCI/hoist/103/1486146609.58/ansible-syntax : SUCCESS in 57s
- http://logs.bonnyci.com/check_github/BonnyCI/hoist/103/1486146609.58/hoist-checks : FAILURE in 12s

**Comment by omgjlk** [comment]
This looks like it'll take our 3 jobs which could be ran in parallel and instead makes it a single serial job?

Where in this new setup is the node label defined, since at least one of these jobs needs the `ubuntu-hoist` node set.

File: tests/test-hoist-ansible.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash -ex
+
+source /opt/venvs/ansible/bin/activate
+
+echo "Running hoist ansible syntax test"
+ansible-playbook -i /dev/null --syntax-check $(find . tests -maxdepth 1 -type f -name \*.yml -not -name .travis.yml)
```
Comment (gandelman-a) [inline tests/test-hoist-ansible.sh:6]: shellchecks SC2046 prohibits this and requires the $() to be quoted, which borks this call as it passes in the entire find result as a single parameter.  Disabled that check below.

File: .bonnyci/run.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+if test -r test-requirements.txt; then
+    pip install -r test-requirements.txt
+fi
+
+if test -r requirements.txt; then
+    pip install -r requirements.txt
+fi
+
+if test -r requirements.yml; then
+    ansible-galaxy install -r install -r requirements.txt
```
Comment (ConsoleCatzirl) [inline .bonnyci/run.sh]: This should be `requirements.yml` instead of `requirements.txt`

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.com/check_github/BonnyCI/hoist/103/1486150139.44/ubuntu-hoist : SUCCESS in 4m 16s
- http://logs.bonnyci.com/check_github/BonnyCI/hoist/103/1486150139.44/ansible-syntax : SUCCESS in 54s
- http://logs.bonnyci.com/check_github/BonnyCI/hoist/103/1486150139.44/hoist-checks : FAILURE in 11s

File: roles/zuul/files/jobs/bonnyci.yaml [hunk 3]
```diff
@@ -10,7 +10,10 @@
               copy-after-failure: true
 
 - job-template:
-    name: 'bonnyci-run-{pipeline}'
+    node: 'ubuntu-xenial'
```
Comment (ConsoleCatzirl) [inline roles/zuul/files/jobs/bonnyci.yaml:13]: Does this work as a default value? I think that this needs to be `node: {node}` in order to use the parameter passed below.

**Review by ConsoleCatzirl — CHANGES_REQUESTED** [review]
For the most part this is looking good

File: .bonnyci/run.sh (hunk 2 above)
Comment (gandelman-a) [reply .bonnyci/run.sh]: doh

File: roles/zuul/files/jobs/bonnyci.yaml (hunk 3 above)
Comment (gandelman-a) [reply roles/zuul/files/jobs/bonnyci.yaml:13]: Yeah, this is just the default value.  Passing a node in below will overwrite it.  Verified this locally by referencing it in the executor and inspecting the generated job.

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.com/check_github/BonnyCI/hoist/103/1486151460.26/ubuntu-hoist : SUCCESS in 4m 14s
- http://logs.bonnyci.com/check_github/BonnyCI/hoist/103/1486151460.26/ansible-syntax : SUCCESS in 55s
- http://logs.bonnyci.com/check_github/BonnyCI/hoist/103/1486151460.26/hoist-checks : FAILURE in 12s

File: roles/zuul/files/jobs/bonnyci.yaml (hunk 3 above)
Comment (ConsoleCatzirl) [reply roles/zuul/files/jobs/bonnyci.yaml:13]: cool

**Review by ConsoleCatzirl — APPROVED** [review]
LGTM

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.com/check_github/BonnyCI/hoist/103/1486424478.52/ubuntu-hoist : SUCCESS in 4m 52s
- http://logs.bonnyci.com/check_github/BonnyCI/hoist/103/1486424478.52/ansible-syntax : SUCCESS in 55s
- http://logs.bonnyci.com/check_github/BonnyCI/hoist/103/1486424478.52/hoist-checks : FAILURE in 11s

**Review by ConsoleCatzirl — APPROVED** [review]

**Comment by omgjlk** [comment]
I'm assuming bonny is not merging this because there are historical negative reviews. Is that right?

**Comment by omgjlk** [comment]
gate-recheck

**Pull request merged**